packages/server/src/integrations/redis.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  DatasourceFieldTypes,
+  Integration,
+  QueryTypes,
+} from "../definitions/datasource"
+import Redis from "ioredis"
+
+module RedisModule {
+  interface RedisConfig {
+    host: string
+    port: number
+    username: string
+    password?: string
+  }
+
+  const SCHEMA: Integration = {
+    docs: "https://redis.io/docs/",
+    description: "",
+    friendlyName: "Redis",
+    datasource: {
+      host: {
+        type: "string",
+        required: true,
+        default: "localhost",
+      },
+      port: {
+        type: "number",
+        required: true,
+        default: 6379,
+      },
+      username: {
+        type: "string",
+        required: false,
+      },
+      password: {
+        type: "password",
+        required: false,
+      },
+    },
+    query: {
+      create: {
+        type: QueryTypes.FIELDS,
+        fields: {
+          key: {
```

Comment:
Should we allow setting a TTL as part of the creation/update? Might be super super useful to be able to set a TTL, if they don't provide a value then just don't set any TTL.

Reply:
Great shout - really nice for caching use cases etc and prevents you needing another delete. Will add!